src/test/java/de/unistuttgart/iste/gits/flashcard_service/api/mutation/query/QueryByAssessmentIdTest.java

```diff
@@ -0,0 +1,120 @@
+package de.unistuttgart.iste.gits.flashcard_service.api.mutation.query;
+
+import de.unistuttgart.iste.gits.common.testutil.GraphQlApiTest;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardEntity;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardSetEntity;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardSideEntity;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.repository.FlashcardSetRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.graphql.test.tester.GraphQlTester;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+@Component
+@GraphQlApiTest
+public class QueryByAssessmentIdTest {
+
+
+    @Autowired
+    private FlashcardSetRepository flashcardSetRepository;
+
+    @Test
+    void testFlashcardSetsByAssessmentIds(GraphQlTester graphQlTester) {
+        // Create test data
+        UUID setId1 = UUID.randomUUID();
+        UUID setId2 = UUID.randomUUID();
+        UUID setId3 = UUID.randomUUID();
+
+        UUID assessmentId1 = UUID.randomUUID();
+        UUID assessmentId2 = UUID.randomUUID();
```

Comment:
It would be more significant test, if you create an assessmentId3 and assign it to flashcardset3. (but keeping the rest as it is, then the expected result is only set1 and set2